swarm/src/lib.rs

```diff
@@ -77,28 +77,33 @@ pub use protocols_handler::{
     SubstreamProtocol
 };
 
+
+pub type BoxSubstream = Substream<StreamMuxerBox>;
+pub type NegotiatedBoxSubstream = Negotiated<BoxSubstream>;
```

Comment:
In light of the previous comment, maybe `pub type NegotiatedSubstream = Negotiated<Substream<StreamMuxerBox>>` could be an alternative?